keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,923 @@
+
+The best way to roll this change back:
+
+1. Configure your container runtime to not split the image filesystem.
+2. Restart the container runtime.
+3. Restart Kubelet with feature flag off.
+
+###### Does enabling the feature change any default behavior?
+
+Yes, we will switch to using `ImageFsInfo` to compute disk stats rather than call `ImageStats`.
+
+The eviction manager will change if the image filesystem is split.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+There are two possibilities for this feature.
+
+1. Container runtime is configured for split disk
+2. Container runtime is not configured for split disk.
+
+If the feature toggle is disabled in 1, then turning off the feature will tell eviction manager that the containerfs=imagefs.  
+The container garbage collection will try to delete the writeable layer on the image filesystem which may not be there.
+Kubelet will still run but there could be a possibility that the container filesystem will grow unchecked and eventually cause disk pressure.
+
+In case 2, rolling back this feature will be possible because we will use `ImageStats` to compute the filesystem usage.
+Since the container runtime is configured to not split the disk, nothing would really be changed in this case.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing as long as the container runtime is set up to split again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, even though roll back is not supported, we will be switching to using `ImageFsInfo` for stats on the file system.
+This will be guarded by a feature gate and we will test negative and positive test cases.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+If the filesystem is not split, this rollout or rollback will be a no-op.
+
+If the filesystem is split and you want to roll back the change that will require a change to the container runtime configuration.
+
+If one does not want to change the container runtime configuration, there could be a possibility that node pressure could happen as garage collection will not work.
+The container filesystem would grow unbounded and would require users to clean up their disks to avoid disk pressure.
+
+###### What specific metrics should inform a rollback?
+
+If a cluster is evicting a lot more pods (`evictions`) than normal, this could be caused by this feature.
```

Comment:
This aspect seems like the main risk, aside from rollbacks requiring cooperation with CRI implementation config (containerd, cri-o, ...), can we outline the potential risk around eviction changes when toggling this feature?

Reply:
I added a paragraph to risk and mitigation about this.  You are correct that this could be the main impact of this change. PTAL here and in the #Risks section.